haystack/core/pipeline/pipeline.py

```diff
@@ -212,6 +212,9 @@ def run(  # noqa: PLR0915, PLR0912, C901, pylint: disable=too-many-branches
         if include_outputs_from is None:
             include_outputs_from = set()
 
+        # track intermediate outputs from components in include_outputs_from to save them to a pipeline snapshot
+        intermediate_outputs: Dict[str, Any] = {}
```

Comment:
So to follow up in our other thread I don't think we need this anymore since we will only save the current state of `pipeline_outputs` down below.